MetaMask/metamask-extension
# feat: use siwe detection from @metamask/controller-utils

## Explanation
This PR aims to remove `detectSIWE` method from the extension code and consume that method from `@metamask/controller-utils`
It also removes usage of `detectSIWE` function from `SignController`, because it's already been implemented in `PersonalMessageManager` to extend message with `siwe` parameters. (See related PR [here](https://github.com/MetaMask/core/pull/1139))

Related issue: 
https://github.com/MetaMask/MetaMask-planning/issues/407

## Manual Testing Steps
- Go [test dapp](https://metamask.github.io/test-dapp/)
- Try "SIGN IN WITH ETHEREUM" button
- Approve approval window 
- See result being printed below buttons

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
